conda_forge_tick/contexts.py

```diff
@@ -40,6 +42,34 @@ def default_branch(self):
         else:
             return self._default_branch
 
-    @default_branch.setter
-    def default_branch(self, v):
-        self._default_branch = v
+    @property
+    def git_repo_name(self) -> str:
+        return f"{self.feedstock_name}-feedstock"
+
+    @contextmanager
+    def reserve_clone_directory(self) -> Iterator[ClonedFeedstockContext]:
+        """
+        Reserve a temporary directory for the feedstock repository that will be available within the context manager.
+        The returned context object will contain the path to the feedstock repository in local_clone_dir.
+        After the context manager exits, the temporary directory will be deleted.
+        """
+        with tempfile.TemporaryDirectory() as tmpdir:

```

Comment:
If I understand the current code correctly, we clone everything into `./feedstocks` relative to the bot code and not into `RUNNER_TEMP`. Also, the `TemporaryDirectory()` context clears the directory automatically without the runner needing to do it (as pointed out in the comment). Also, the tests using the temporary directory all run though on GitHub Actions.

Since I also could not find any related issues with a quick Google search, do you remember what exactly the issue has been with the normal temp file location?

If we figure that we want to use `RUNNER_TEMP`, I think the best solution is to [set the `TMPDIR` environment variable](https://docs.python.org/3/library/tempfile.html#tempfile.gettempdir) to the value of `RUNNER_TEMP`.

But, before we do so, I would prefer seeing a clear indication to see why it is necessary, since there is the KISS principle.

Cc @0xbe7a - maybe you have experience with GHA temp file locations?

Reply:
Using the feedstocks dir in the work area is a way to ensure we have the disk space iirc. I'd rather use runner temp here from the start than deal with the fallout later.

Reply:
Implemented in bc6f5ce